forge-ai/src/main/java/forge/ai/ComputerUtilCost.java

```diff
@@ -358,11 +366,13 @@ public static boolean checkSacrificeCost(final Player ai, final Cost cost, final
                     }
                     if (source.isCreature()) {
                         // e.g. Sakura Tribe-Elder
+                        final Combat combat = ai.getGame().getCombat();
                         final boolean beforeNextTurn = ai.getGame().getPhaseHandler().is(PhaseType.END_OF_TURN) && ai.getGame().getPhaseHandler().getNextTurn().equals(ai);
-                        final boolean creatureInDanger = ComputerUtil.predictCreatureWillDieThisTurn(ai, source, sourceAbility);
+                        final boolean creatureInDanger = ComputerUtil.predictCreatureWillDieThisTurn(ai, source, sourceAbility, false)
+                                && !ComputerUtilCombat.willOpposingCreatureDieInCombat(ai, source, combat);
                         final int lifeThreshold = (((PlayerControllerAi) ai.getController()).getAi().getIntProperty(AiProps.AI_IN_DANGER_THRESHOLD));
```

Comment:
this here is another unsafe cast which will crash it on certain paths
I think we really need to come up with some clean solution to this as follow-up, this is turning into quite the patchwork with very difficult to debug outcomes

Reply:
so computerutil is used by automation/prediction even the controller is not AI? there should be a check on the controller on every method that casts PlayerControllerAI to prevent unsafe cast

Reply:
E. g. those static booleans are rather problematic because they're obviously shared across the whole application

this has the side effect of "breaking contract" for independent game threads since AI thinking in Game 1 could toggle the field for AI in Game 2 at an unfortunate time
- now most users might not have two games active at the same time but how do tournaments get played out in the background?
- I'm also not 100% sure such race conditions couldn't already happen in a single game with more than one AiController

Reply:
hmm I thought the method call is per Game, so AIController is shared? that's not good since it will affect the other game at least for desktop version.

Reply:
`PlayerControllerAi + AiController` are exclusive objects for each Game and AI player
but you can always add static variables which by definition exist only once and then behave as a poor version of information sharing between classes

Reply:
Yeah, agreed, though solving this globally goes beyond the scope of this MR, I'm afraid (and may require quite a bit of planning and assistance, too). I'll add a safety check here and we can think as to how to solve the bigger issue as a follow-up task :)

Reply:
Done :)